src/ansys/fluent/core/launcher/fluent_container.py

```diff
@@ -239,16 +239,23 @@ def configure_container_dict(
     logger.warning(
         f"Starting Fluent container mounted to {mount_source}, with this path available as {mount_target} for the Fluent session running inside the container."
     )
-
-    if not port and "ports" in container_dict:
+    port_mapping = {port: port} if port else {}
+    if not port_mapping and "ports" in container_dict:
         # take the specified 'port', OR the first port value from the specified 'ports', for Fluent to use
-        port = next(iter(container_dict["ports"].values()))
-    if not port and pyfluent.LAUNCH_FLUENT_PORT:
+        port_mapping = container_dict["ports"]
+    if not port_mapping and pyfluent.LAUNCH_FLUENT_PORT:
         port = pyfluent.LAUNCH_FLUENT_PORT
-    if not port:
+        port_mapping = {port: port}
+    if not port_mapping:
         port = get_free_port()
+        port_mapping = {port: port}
 
-    container_dict.update(ports={str(port): port})  # container port : host port
+    container_dict.update(
+        ports={str(x): y for x, y in port_mapping.items()}
+    )  # container port : host port
+    container_grpc_port = next(
+        iter(port_mapping.values())
+    )  # the first port in the mapping is chosen as the gRPC port
```

Comment:
@lagacep-ans If multiple ports are sent, the first port will be chosen as the grpc port.